elements/notification-container/notification-container.js

```diff
@@ -0,0 +1,86 @@
+import BulbsElement from 'bulbs-elements/bulbs-element';
+import NotificationDisplay from './notification-display';
+import invariant from 'invariant';
+import React, { PropTypes } from 'react';
+import { registerReactElement } from 'bulbs-elements/register';
+
+
+var localStorageMixin = require('react-localstorage');
+var reactMixin = require('react-mixin');
+
+
+class NotificationContainer extends BulbsElement {
+
+  constructor (props) {
+    invariant(!!props.src, 'notification-container component requires a src');
+    super(props);
+    this.state = {
+      next: null,
+      local_notification_ids: [],
+      notification: {}
+    };
+  }
+
+  getStateFilterKeys () {
+    return ['local_notification_ids'];
+  }
+
+  requestNotifications (src) {
+    fetch(src)
+      .then(function(response) {
+        return response.json();
+      })
+      .then(this.handleRequestSuccess.bind(this));
+  }
+
+  componentDidMount () {
+    this.requestNotifications(this.props.src);
+  }
+
+  handleRequestSuccess (response) {
+    console.log(response);
+    var notifications = response.results;
+    this.setState({ next: response.next });
+
+    if (this.state.local_notification_ids.length > 0) {
+      notifications = this.removeLocalNotifications(notifications);
+    }
+
+    if (notifications.length > 0) {
+      this.setState({ notification: notifications[0] });
+      this.setState({
+        local_notification_ids: this.state.local_notification_ids.concat([notifications[0].id])
+      });;
+    } else if (this.state.next !== null) {
+      this.requestNotifications(this.state.next);
+    }
+
+  }
+
+  removeLocalNotifications (notifications) {
+    for (var i = notifications.length - 1; i >= 0; i--) {
+      if (this.state.local_notification_ids.indexOf(notifications[i].id) != -1) {
+        notifications.splice(i, 1);
+      }
+    }
+    return notifications;
+  }
```

Comment:
We use `foreach` and then for style, here I would create a new array, and build it up rather than mutate the passed-in array.

```js
var withoutLocal = [];
notifications.forEach((notification) => {
  if (/* conditional */) {
    withoutLocal.push(notification);
  }
});
return withoutLocal;
```